### ISSUE TITLE
PM UI:  cannot bulk update packages

### ISSUE THREAD

**Issue by dtivel** [NuGet/Home#10510]
## Details about Problem

https://github.com/NuGet/Home/issues/10510 regressed PM UI update scenarios.  The root problem is [this](https://github.com/NuGet/NuGet.Client/commit/c995174aaea8d55e1504e9afba83696922497712#diff-80991ce1edef4f242608751b70dd2d48dad8e24a28dbefd2b68a3a7784a12f21R1556) change.  Previously, updating selected packages in the Updates tab of PM UI passes the package identities for the _desired_ packages.  This commit passes the package identities for the _currently installed_ packages.  As a result, PM UI now reports that are no actions to perform.

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe):  Visual Studio PM UI

### Product version: 
Microsoft Visual Studio Enterprise 2019 Int Preview
Version 16.10.0 Preview 1.0 [30929.219.main]
NuGet:  latest dev (@ commit 4991066d8d78053b389b07819afba78b88e66340)

Worked before? If so, with which NuGet version:   dev 5.9.0.7083

## Repro steps

1.  Create a new C# ASP.NET Web Application (.NET Framework) with these options:

    *  Framework:  .NET Framework 4.8
    *  Type:  Web Forms

2.  In Solution Explorer right-click on the project node and select "Manage NuGet Packages..."
3.  In NuGet Package Manager UI navigate to the Updates tab.
4.  Check (enable) the "Select all packages" checkbox.
5.  Click the Update button.

### Expected result
Error List:  "Unable to resolve dependencies. 'bootstrap 5.0.0-beta1' is not compatible with 'AspNet.ScriptManager.bootstrap 4.5.3 constraint: bootstrap (>= 4.5.2 && < 4.5.4)'.	"		

### Actual result
*  Error List:  "An internal error occurred. Please contact Microsoft Support."	
*  Output window's "Package Manager" pane:

```
Attempting to gather dependency information for multiple packages with respect to project 'WebApplication1', targeting '.NETFramework,Version=v4.8'
Gathering dependency information took 827 ms
Attempting to resolve dependencies for multiple packages.
Resolving dependency information took 0 ms
Resolving actions install multiple packages
Resolution was successful but resulted in no action
There are no new updates available.
No package updates are available from the current package source for project 'WebApplication1'.
StreamJsonRpc.RemoteInvocationException: An internal error occurred. Please contact Microsoft Support.
   at Microsoft.ServiceHub.Framework.ServiceJsonRpcDescriptor.LocalProxyGeneration.<ReturnedValueTaskOfTHelper>d__25`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()
   at NuGet.PackageManagement.UI.UIActionEngine.<ShouldContinueDueToDotnetDeprecationAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ValueTaskAwaiter`1.GetResult()
   at NuGet.PackageManagement.UI.UIActionEngine.<>c__DisplayClass11_0.<<PerformActionImplAsync>b__0>d.MoveNext()
RPC server exception:
Microsoft.Assumes+InternalErrorException: An internal error occurred. Please contact Microsoft Support.
      at Microsoft.Assumes.Fail(String message)
      at Microsoft.Assumes.True(Boolean condition, String message)
      at Microsoft.Assumes.NotNullOrEmpty[T](ICollection`1 values)
      at NuGet.PackageManagement.VisualStudio.NuGetProjectManagerService.<GetProjectsWithDeprecatedDotnetFrameworkAsync>d__22.MoveNext()
   --- End of stack trace from previous location where exception was thrown ---
      at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      at System.Threading.Tasks.ValueTask`1.get_Result()
      at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()
      at Microsoft.ServiceHub.Framework.ServiceJsonRpcDescriptor.LocalProxyGeneration.<ReturnedValueTaskOfTHelper>d__25`1.MoveNext()

Time Elapsed: 00:00:01.0345980
========== Finished ==========
```

### Verbose Logs

**Comment by donnie-msft** [comment]
@dtivel thanks for reporting this. So far, I tried your repro and am getting something different. Maybe the scenario is specific. 
Are you connected to a CodeSpace or anything else of note?
![image](https://user-images.githubusercontent.com/49205731/106478239-fee0dc80-6476-11eb-83f4-fefba7f4465c.png)

**Comment by donnie-msft** [comment]
Found the repro. You have to load the Installed tab first, then switch to Updates. It won't have the latest loaded in that scenario.
If you open PMUI directly to Updates, or Browse tab then switch to Updates, this won't repro.

**Comment by donnie-msft** [comment]
Released in 16.9 Preview 5
